### PR TITLE
Bug 1223733 - Migrating Pin page dialog into web components r=mhenretty

### PR DESCRIPTION
**Pull request by albertopq**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=15054921b8a2b72bab91c06325a7ef7a7be66eff

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=beb1e898cae907244a1e50ff65dfe939e1d53a72

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=83f275a3fdff7a47abe203950873f1cf43bd8160

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=6921deae94d838360bfdb8508d3d31193c38a134

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=078c240d62fd01c686920e3eba2fcba3f10adac8

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=d17258f364fe816e9b62e61cc5d0865e2d9a7a5a

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=944eb6fdb001a5ba02cd4ddeda4ea9f98d4e8b0c

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=3522693909eead37fe22e6303b9a7353b2195228

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=8334a80c817db24745d244c7311cae69ac35d673

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=40eaf8ce94f44071de6a63d5c405795760d3aa57

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=ac434d534d2b03a64a6813fd4ca434cbbf2b4790

File: apps/system/js/app_window_manager.js [hunk 1]
```diff
@@ -118,7 +118,7 @@
      * @type {Boolean}
      * @memberOf AppWindowManager
      */
-    slowTransition: false,
+    slowTransition: true,
```
Comment (mikehenrty) [inline apps/system/js/app_window_manager.js]: nit: undo

File: apps/system/js/pin_page_system_dialog.js [hunk 2]
```diff
@@ -12,12 +12,14 @@
...
     this.options = {};
-    this.render();
     LazyLoader.load('shared/elements/gaia-component/gaia-component.js')
       .then(() => {
         LazyLoader.load('shared/elements/gaia-site-icon/script.js');
+        LazyLoader.load('shared/elements/gaia-button/gaia-button.js');
+        LazyLoader.load('/shared/elements/gaia-dialog/gaia-dialog.js');
         LazyLoader.load('shared/elements/gaia-toast/gaia-toast.js')
           .then(() => {
+            this.render();
```
Comment (mikehenrty) [inline apps/system/js/pin_page_system_dialog.js]: This looks racy to me. We aren't waiting for gaia-button and gaia-dialog to load before rendering the entire dialog. I think we to wait for those two components to load (perhaps using Promise.all) before calling render.

File: apps/system/js/pin_page_system_dialog.js [hunk 3]
```diff
@@ -135,10 +147,7 @@
         break;
     }
 
-    // Waiting for the animation. We should migrate this to a
-    // hide event when https://github.com/gaia-components/gaia-toast/issues/2
-    // gets fixed
-    setTimeout(this.close.bind(this), this._banner.timeout);
+    this.element.close();
```
Comment (mikehenrty) [inline apps/system/js/pin_page_system_dialog.js]: :+1:

File: apps/system/js/pin_page_system_dialog.js [hunk 4]
```diff
@@ -203,6 +215,24 @@
...
+    // as it is fixed positioned
+    this.element.style.height = height + 'px';
+    this.debug('updateHeight: new height = ' + height);
+  };
+
+  SystemDialog.prototype.updateWidth = function sd_updateWidth() {
+    var width = Service.query('LayoutManager.width');
+    this.containerElement.style.width = width + 'px';
+    // We need to manually set the gaia-dialog width,
+    // as it is fixed positioned
```
Comment (mikehenrty) [inline apps/system/js/pin_page_system_dialog.js]: Even with fixed, we can't use `100%` here?

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=fa94260f87a187869d95a844343654dff0043dda

File: apps/system/js/pin_page_system_dialog.js (hunk 4 above)
Comment (albertopq) [inline apps/system/js/pin_page_system_dialog.js]: The whole problem about fixed and system dialogs is the SHB. 100% would mean the dialog overlaying the SHB in both portrait and landscape mode.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=af4ab69f41c8b631401750b3a33a9f125ffe497b

File: apps/system/js/pin_page_system_dialog.js (hunk 4 above)
Comment (mikehenrty) [inline apps/system/js/pin_page_system_dialog.js]: ah right, i forgot about landscape.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=0103a5eedecced748f3411d3bc3f52d1a5395c9f

**Pull request merged**